pkg/operation/botanist/addons.go

```diff
@@ -233,8 +233,21 @@ func (b *Botanist) generateCoreAddonsChart() (*chartrenderer.RenderedChart, erro
 		nodeExporterConfig        = map[string]interface{}{}
 		blackboxExporterConfig    = map[string]interface{}{}
 		nodeProblemDetectorConfig = map[string]interface{}{}
+		networkPolicyConfig       = map[string]interface{}{}
 	)
 
+	k8sEndpoint := &corev1.Endpoints{}
+	if err := b.K8sShootClient.Client().Get(context.TODO(), client.ObjectKey{Name: "kubernetes", Namespace: "default"}, k8sEndpoint); err != nil {
+		return nil, err
+	}
+	endpointIps := []string{}
+	for _, ep := range k8sEndpoint.Subsets {
```

Comment:
the only concern here is that if the endpoints change, the black-box exporter won't be able to talk to the API server due to wrong policy. This can lead to false-positive in the dashboard implying the shoot can't reach seed.

Reply:
What do you recommend we do? If there are network policies from users then the we also don't have any metrics since the blackbox exporter is not reachable. Theoretically we could allow all egress to `0.0.0.0/0`. Not sure if this is an acceptable solution.

Reply:
Yes, I'd vote for egress `0.0.0.0/0` for the blackbox-exporter.

Reply:
Changed the egress policy to `0.0.0.0/0`. PTAL

Reply:
for other providers we can actually use the IPs since they are fixed, for for now lets go with `0.0.0.0/0`